atom/src/validatebehavior.cpp

```diff
@@ -244,10 +244,12 @@ long_handler( Member* member, CAtom* atom, PyObject* oldvalue, PyObject* newvalu
 static PyObject*
 long_promote_handler( Member* member, CAtom* atom, PyObject* oldvalue, PyObject* newvalue )
 {
-    if( PyLong_Check( newvalue ) )
+    if( Py23Int_Check( newvalue ) )
         return newref( newvalue );
-    if( PyInt_Check( newvalue ) )
-        return PyLong_FromLong( PyInt_AS_LONG( newvalue ) );
+    #if PY_MAJOR_VERSION < 3
+        if( PyInt_Check( newvalue ) )
+            return PyInt_FromLong( PyInt_AS_LONG( newvalue ) );
+    #endif
```

Comment:
You're leaving this as Python 2 only, but you appear to be changing the behavior.